src/listoper.c

```diff
@@ -1579,33 +1579,55 @@ Obj FuncADD_ROW_VECTOR_2_FAST ( Obj self,
 
 /****************************************************************************
 **
-*F  FuncMULT_ROW_VECTOR_2( <self>, <list>, <mult> )
+*F  FuncMULT_VECTOR_LEFT_RIGHT_2( <self>, <list>, <mult> )
 **
-**  This function destructively multiplies the entries of <list> by <mult>
-**  It does very little checking
+**  This function destructively multiplies the entries of <list> by <mult>.
+**  It multiplies with <mult> from the left if <left> is not 0 and from the
+**  right otherwise.
+**  It does very little checking.
 **
 */
 
-Obj FuncMULT_ROW_VECTOR_2( Obj self,
+Obj FuncMULT_VECTOR_LEFT_RIGHT_2( Obj self,
                            Obj list,
-                           Obj mult )
+                           Obj mult,
+                           UInt left )
 {
   UInt i;
   Obj prd;
   UInt len = LEN_LIST(list);
-  for (i = 1; i <= len; i++)
-    {
-      prd = ELMW_LIST(list,i);
-      prd = PROD(prd,mult);
-      ASS_LIST(list,i,prd);
-      CHANGED_BAG(list);
-    }
+  if (left != 0)
+    for (i = 1; i <= len; i++)
+      {
+        prd = ELMW_LIST(list,i);
+        prd = PROD(mult,prd);
+        ASS_LIST(list,i,prd);
+        CHANGED_BAG(list);
+      }
+  else
+    for (i = 1; i <= len; i++)
+      {
+        prd = ELMW_LIST(list,i);
+        prd = PROD(prd,mult);
+        ASS_LIST(list,i,prd);
+        CHANGED_BAG(list);
+      }
```

Comment:
Do you have `clang-format` installed? If so, please consider using `git clang-format` on your commits to reformat the modified code.

But don't worry about this if you don't have `clang-format`.